### PR TITLE
refactor: gracefully exit ticker

### PR DESCRIPTION
**Pull request by rhnvrm**
Ref: #28

### CODE REVIEW

File: ticker/ticker.go [hunk 1]
```diff
@@ -426,6 +459,16 @@ func (t *Ticker) Close() error {
 	return t.Conn.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
 }
 
+// Stop the ticker instance and all the goroutines it has spawned.
+func (t *Ticker) Stop() error {
+	if t.cancel == nil {
+		return fmt.Errorf("ticker is not running")
```
Comment (vividvilla) [inline ticker/ticker.go:465]: Why do we have to raise error if its not running? in that case `Stop()` doesn't need to return anything.

**Review by vividvilla — COMMENTED** [review]
LGTM apart from that one comment on `Stop()` method.

**Comment by abhirathmahipal** [comment]
@rhnvrm Seems to work fine. Just tested it out.

Have attached two screenshots.
- Your code on my local machine.
- Refreshing connections in the same process without exiting (have blurred out my app key, secret and access token)
- Getting live ticks again.

image
image
image

**Comment by abhirathmahipal** [comment]
@rhnvrm ICYMI ^

**Pull request merged**